### PR TITLE
[dx12] Fix partial texture barrier not affecting stencil aspect

### PR DESCRIPTION
**Pull request by Wumpf**
**Connections**
Issues accidentally covered by tests in #2307 (i.e. this change is required to land it)

**Description**
Fix partial texture barrier (i.e. when not the entire texture is transitioned) not affecting stencil aspect.
Fix clearing of D24Plus also clearing "hidden" stencil.

**Testing**
Tested in #2307, a dedicated test would be nice to be fair but this will have to do for now

### CODE REVIEW

File: wgpu-hal/src/dx12/command.rs [hunk 1]
```diff
@@ -319,15 +321,29 @@ impl crate::CommandEncoder<super::Api> for super::CommandEncoder {
                     // Only one barrier if it affects the whole image.
                     self.temp.barriers.push(raw);
                 } else {
-                    // Generate barrier for each layer/level combination.
+                    let has_stencil = FormatAspects::from(barrier.texture.format)
+                        .contains(FormatAspects::STENCIL);
+                    let planes = if has_stencil {
+                        match barrier.range.aspect {
+                            wgt::TextureAspect::All => 0..2,
+                            wgt::TextureAspect::StencilOnly => 1..2,
```
Comment (kvark) [inline wgpu-hal/src/dx12/command.rs:329]: what if it's a stencil-only format? would it still be 1..2?

File: wgpu-hal/src/dx12/command.rs [hunk 2]
```diff
@@ -1,3 +1,5 @@
+use crate::FormatAspects;
```
Comment (kvark) [inline wgpu-hal/src/dx12/command.rs:1]: let's not diverge from the current style where `crate::xxx` are either used directly, or included close to the place they are used

File: wgpu-hal/src/dx12/command.rs [hunk 3]
```diff
@@ -610,7 +626,9 @@ impl crate::CommandEncoder<super::Api> for super::CommandEncoder {
             if !ds.depth_ops.contains(crate::AttachmentOps::LOAD) {
                 flags |= native::ClearFlags::DEPTH;
             }
-            if !ds.stencil_ops.contains(crate::AttachmentOps::LOAD) {
+            if !ds.stencil_ops.contains(crate::AttachmentOps::LOAD)
+                && ds.target.view.has_stencil_aspect
```
Comment (kvark) [inline wgpu-hal/src/dx12/command.rs:630]: similarly, if we are doing this check for stencil, we need to do a similar one for depth

**Review by kvark — CHANGES_REQUESTED** [review]

File: wgpu-hal/src/dx12/command.rs (hunk 2 above)
Comment (Wumpf) [reply wgpu-hal/src/dx12/command.rs]: `use` and me really don't get along well 😄

File: wgpu-hal/src/dx12/command.rs (hunk 1 above)
Comment (Wumpf) [reply wgpu-hal/src/dx12/command.rs:328]: huh, yeah that should be 0..1 as well I suppose! Fixed!

**Review by kvark — APPROVED** [review]

**Comment by kvark** [comment]
published in wgpu-hal-0.12.1

**Pull request merged**